### PR TITLE
[macOS] Use NSAlert APIs to show alert suppression checkbox

### LINKED ISSUE #3332: [macOS] Make Quit confirmation alert optional even with active transfers.
RIght now if you try to close Transmission while there's any active transfer, confirmation alert is shown.
It's better to leave user an option to skip it.
By default it should be enabled, but it could be disabled in Preferences or with checkbox right inside the alert itself.

CC @ckerr @sweetppro @ckerr @bacecek

### PR DESCRIPTION
**Pull request by nevack**
Fixes #3332

Screenshot 2022-06-27 at 03 11 39

### CODE REVIEW

**Comment by mangosplat** [comment]
Thanks.
Would be nice if there was a way to re-enable alerts in preferences

**Comment by nevack** [comment]
> Thanks. Would be nice if there was a way to re-enable alerts in preferences

It's already there.

Screenshot 2022-06-27 at 02 22 54

**Review by sweetppro — APPROVED** [review]
looks great, and works well

File: macosx/Controller.mm [hunk 1]
```diff
@@ -868,47 +868,57 @@ - (BOOL)applicationShouldHandleReopen:(NSApplication*)app hasVisibleWindows:(BOO
...
+                                       "Confirm Quit panel -> message"),
+                                   active];
+    [alert addButtonWithTitle:NSLocalizedString(@"Quit", "Confirm Quit panel -> button")];
+    [alert addButtonWithTitle:NSLocalizedString(@"Cancel", "Confirm Quit panel -> button")];
+    alert.showsSuppressionButton = YES;
+
+    [alert beginSheetModalForWindow:self.fWindow completionHandler:^(NSModalResponse returnCode) {
+        if (alert.suppressionButton.state == NSControlStateValueOn)
+        {
+            [self.fDefaults setBool:NO forKey:@"CheckQuit"];
```
Comment (DevilDimon) [inline macosx/Controller.mm:916]: I might be wrong here, but won't it cause a retain cycle here? I'm thinking self -> window -> alert -> block -> self

Comment (nevack) [reply macosx/Controller.mm:916]: Is there way to test it properly?
Is there's a bug, do you know, how to fix it?
One easy fix I know is to use `NSUserDefaults.standardUserDefaults` directly here.

**Comment by nevack** [comment]
@ckerr What's the status of this PR?
I think it's ready to merge.

**Comment by ckerr** [comment]
I was unsure since @DevilDimon didn't mark the earlier discussion as resolved

**Pull request merged**